lib/puppet/provider/elastic_stack_keystore/ruby.rb

```diff
@@ -0,0 +1,345 @@
+# frozen_string_literal: true
+
+Puppet::Type.type(:elastic_stack_keystore).provide(
+  :elastic_stack_keystore
+) do
+  desc 'Provider for both `elasticsearch-keystore` and `kibana-keystore` based secret management.'
+
+  mk_resource_methods
+
+  def self.defaults_dir
+    @defaults_dir ||= case Facter.value('osfamily')
```

Comment:
please don't use the legacy fact osfamily, that's deprecated. Please use the nested `os.family`.